### PR TITLE
Various debug fixes

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

* Fix regression showing threads `rip` for `dpt` on Windows
* Improve `dpt` output:

    * Consider flags too when printing thread address
    * Improve `RzCoreBind->getNameDelta` to use realname if `flags->realnames` is true
    * Fix negative offsets
* Fix flagspace selection side-effects after calling `rz_core_bin_apply` APIs
* Fix wrong register profile after attaching to a process with `dpa`
* Fix wrong base address when opening a process memory
* Fix wrong register profile after opening a process memory

**Closing issues**

None

### CODE REVIEW

**Comment by XVilka** [comment]
Somehow it breaks two `rz-diff` tests. cc @wargio:

The new output might be actually better, just need to be checked

https://app.travis-ci.com/github/rizinorg/rizin/jobs/526795256#L3674

```
[XX] db/tools/rz_diff rz-diff functions comparison JSON
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc '!rz-diff -C -jt functions bins/other/rz-diff/true bins/other/rz-diff/false' =
-- stdout
--- expected
+++ actual
@@ -1,1 +1,1 @@
-[{"distance":1.000000,"type":"MATCH","original":{"name":"fcn.00401050","addr":4198480,"size":26},"modified":{"name":"fcn.00401050","addr":4198480,"size":26}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__uflow","addr":4198528,"size":6},"modified":{"name":"sym.imp.__uflow","addr":4198528,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.getenv","addr":4198544,"size":6},"modified":{"name":"sym.imp.getenv","addr":4198544,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.free","addr":4198560,"size":6},"modified":{"name":"sym.imp.free","addr":4198560,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.abort","addr":4198576,"size":6},"modified":{"name":"sym.imp.abort","addr":4198576,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__errno_location","addr":4198592,"size":6},"modified":{"name":"sym.imp.__errno_location","addr":4198592,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.strncmp","addr":4198608,"size":6},"modified":{"name":"sym.imp.strncmp","addr":4198608,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp._exit","addr":4198624,"size":6},"modified":{"name":"sym.imp._exit","addr":4198624,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.strcpy","addr":4198640,"size":6},"modified":{"name":"sym.imp.strcpy","addr":4198640,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__fpending","addr":4198656,"size":6},"modified":{"name":"sym.imp.__fpending","addr":4198656,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.textdomain","addr":4198672,"size":6},"modified":{"name":"sym.imp.textdomain","addr":4198672,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.bindtextdomain","addr":4198704,"size":6},"modified":{"name":"sym.imp.bindtextdomain","addr":4198704,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.dcgettext","addr":4198720,"size":6},"modified":{"name":"sym.imp.dcgettext","addr":4198720,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__ctype_get_mb_cur_max","addr":4198736,"size":6},"modified":{"name":"sym.imp.__ctype_get_mb_cur_max","addr":4198736,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.strlen","addr":4198752,"size":6},"modified":{"name":"sym.imp.strlen","addr":4198752,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__stack_chk_fail","addr":4198768,"size":6},"modified":{"name":"sym.imp.__stack_chk_fail","addr":4198768,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.strrchr","addr":4198800,"size":6},"modified":{"name":"sym.imp.strrchr","addr":4198800,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.lseek","addr":4198816,"size":6},"modified":{"name":"sym.imp.lseek","addr":4198816,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.memset","addr":4198832,"size":6},"modified":{"name":"sym.imp.memset","addr":4198832,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.fscanf","addr":4198848,"size":6},"modified":{"name":"sym.imp.fscanf","addr":4198848,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.close","addr":4198864,"size":6},"modified":{"name":"sym.imp.close","addr":4198864,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__libc_start_main","addr":4198880,"size":6},"modified":{"name":"sym.imp.__libc_start_main","addr":4198880,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.memcmp","addr":4198896,"size":6},"modified":{"name":"sym.imp.memcmp","addr":4198896,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.fputs_unlocked","addr":4198912,"size":6},"modified":{"name":"sym.imp.fputs_unlocked","addr":4198912,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.calloc","addr":4198928,"size":6},"modified":{"name":"sym.imp.calloc","addr":4198928,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.strcmp","addr":4198944,"size":6},"modified":{"name":"sym.imp.strcmp","addr":4198944,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"loc.imp.__gmon_start","addr":4198960,"size":6},"modified":{"name":"loc.imp.__gmon_start","addr":4198960,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.memcpy","addr":4198976,"size":6},"modified":{"name":"sym.imp.memcpy","addr":4198976,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.fileno","addr":4198992,"size":6},"modified":{"name":"sym.imp.fileno","addr":4198992,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.malloc","addr":4199008,"size":6},"modified":{"name":"sym.imp.malloc","addr":4199008,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.nl_langinfo","addr":4199040,"size":6},"modified":{"name":"sym.imp.nl_langinfo","addr":4199040,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.ungetc","addr":4199056,"size":6},"modified":{"name":"sym.imp.ungetc","addr":4199056,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__freading","addr":4199072,"size":6},"modified":{"name":"sym.imp.__freading","addr":4199072,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.realloc","addr":4199088,"size":6},"modified":{"name":"sym.imp.realloc","addr":4199088,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.fdopen","addr":4199104,"size":6},"modified":{"name":"sym.imp.fdopen","addr":4199104,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.setlocale","addr":4199120,"size":6},"modified":{"name":"sym.imp.setlocale","addr":4199120,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__printf_chk","addr":4199136,"size":6},"modified":{"name":"sym.imp.__printf_chk","addr":4199136,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.error","addr":4199152,"size":6},"modified":{"name":"sym.imp.error","addr":4199152,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.open","addr":4199168,"size":6},"modified":{"name":"sym.imp.open","addr":4199168,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.exit","addr":4199216,"size":6},"modified":{"name":"sym.imp.exit","addr":4199216,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.fwrite","addr":4199232,"size":6},"modified":{"name":"sym.imp.fwrite","addr":4199232,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__fprintf_chk","addr":4199248,"size":6},"modified":{"name":"sym.imp.__fprintf_chk","addr":4199248,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.mbsinit","addr":4199264,"size":6},"modified":{"name":"sym.imp.mbsinit","addr":4199264,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.iswprint","addr":4199280,"size":6},"modified":{"name":"sym.imp.iswprint","addr":4199280,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__ctype_b_loc","addr":4199296,"size":6},"modified":{"name":"sym.imp.__ctype_b_loc","addr":4199296,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"fcn.00401460","addr":4199520,"size":41},"modified":{"name":"fcn.00401470","addr":4199536,"size":41}},{"distance":0.879195,"type":"SIMILAR","original":{"name":"fcn.00401530","addr":4199728,"size":745},"modified":{"name":"fcn.00401540","addr":4199744,"size":745}},{"distance":0.931973,"type":"SIMILAR","original":{"name":"fcn.004018f0","addr":4200688,"size":147},"modified":{"name":"fcn.00401900","addr":4200704,"size":147}},{"distance":0.985401,"type":"SIMILAR","original":{"name":"fcn.00401990","addr":4200848,"size":137},"modified":{"name":"fcn.004019a0","addr":4200864,"size":137}},{"distance":1.000000,"type":"MATCH","original":{"name":"fcn.00401a20","addr":4200992,"size":162},"modified":{"name":"fcn.00401a30","addr":4201008,"size":162}},{"distance":0.964912,"type":"SIMILAR","original":{"name":"fcn.00401ae0","addr":4201184,"size":228},"modified":{"name":"fcn.00401af0","addr":4201200,"size":228}},{"distance":0.991569,"type":"SIMILAR","original":{"name":"fcn.00401be0","addr":4201440,"size":2728},"modified":{"name":"fcn.00401bf0","addr":4201456,"size":2728}},{"distance":0.971765,"type":"SIMILAR","original":{"name":"fcn.00402730","addr":4204336,"size":425},"modified":{"name":"fcn.00402740","addr":4204352,"size":425}},{"distance":0.980000,"type":"SIMILAR","original":{"name":"fcn.004029a0","addr":4204960,"size":50},"modified":{"name":"fcn.004029b0","addr":4204976,"size":50}},{"distance":0.960784,"type":"SIMILAR","original":{"name":"fcn.00402e20","addr":4206112,"size":204},"modified":{"name":"fcn.00402e30","addr":4206128,"size":204}},{"distance":0.952331,"type":"SIMILAR","original":{"name":"fcn.00402fc0","addr":4206528,"size":944},"modified":{"name":"fcn.00402fd0","addr":4206544,"size":944}},{"distance":0.992366,"type":"SIMILAR","original":{"name":"fcn.004033d0","addr":4207568,"size":131},"modified":{"name":"fcn.004033e0","addr":4207584,"size":131}},{"distance":0.994012,"type":"SIMILAR","original":{"name":"fcn.00403460","addr":4207712,"size":167},"modified":{"name":"fcn.00403470","addr":4207728,"size":167}},{"distance":0.961538,"type":"SIMILAR","original":{"name":"fcn.00403590","addr":4208016,"size":26},"modified":{"name":"fcn.004035a0","addr":4208032,"size":26}},{"distance":0.958333,"type":"SIMILAR","original":{"name":"fcn.004035e0","addr":4208096,"size":48},"modified":{"name":"fcn.004035f0","addr":4208112,"size":48}},{"distance":0.975000,"type":"SIMILAR","original":{"name":"fcn.00403750","addr":4208464,"size":40},"modified":{"name":"fcn.00403760","addr":4208480,"size":40}},{"distance":0.857143,"type":"SIMILAR","original":{"name":"fcn.004037a0","addr":4208544,"size":49},"modified":{"name":"fcn.004037b0","addr":4208560,"size":49}},{"distance":0.926829,"type":"SIMILAR","original":{"name":"fcn.004037e0","addr":4208608,"size":41},"modified":{"name":"fcn.004037f0","addr":4208624,"size":41}},{"distance":1.000000,"type":"MATCH","original":{"name":"fcn.00403810","addr":4208656,"size":67},"modified":{"name":"fcn.00403820","addr":4208672,"size":67}},{"distance":0.972222,"type":"SIMILAR","original":{"name":"fcn.00403860","addr":4208736,"size":108},"modified":{"name":"fcn.00403870","addr":4208752,"size":108}},{"distance":0.958621,"type":"SIMILAR","original":{"name":"fcn.004038e0","addr":4208864,"size":1015},"modified":{"name":"fcn.004038f0","addr":4208880,"size":1015}},{"distance":0.935484,"type":"SIMILAR","original":{"name":"fcn.00403cf0","addr":4209904,"size":124},"modified":{"name":"fcn.00403d00","addr":4209920,"size":124}},{"distance":0.953846,"type":"SIMILAR","original":{"name":"fcn.00403d70","addr":4210032,"size":65},"modified":{"name":"fcn.00403d80","addr":4210048,"size":65}},{"distance":0.967391,"type":"SIMILAR","original":{"name":"fcn.00403db0","addr":4210096,"size":92},"modified":{"name":"fcn.00403dc0","addr":4210112,"size":92}},{"distance":0.937500,"type":"SIMILAR","original":{"name":"fcn.00403e90","addr":4210320,"size":32},"modified":{"name":"fcn.00403ea0","addr":4210336,"size":32}}]

+[{"distance":1.000000,"type":"MATCH","original":{"name":"fcn.00401050","addr":4198480,"size":26},"modified":{"name":"fcn.00401050","addr":4198480,"size":26}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__uflow","addr":4198528,"size":6},"modified":{"name":"sym.imp.__uflow","addr":4198528,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.getenv","addr":4198544,"size":6},"modified":{"name":"sym.imp.getenv","addr":4198544,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.free","addr":4198560,"size":6},"modified":{"name":"sym.imp.free","addr":4198560,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.abort","addr":4198576,"size":6},"modified":{"name":"sym.imp.abort","addr":4198576,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__errno_location","addr":4198592,"size":6},"modified":{"name":"sym.imp.__errno_location","addr":4198592,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.strncmp","addr":4198608,"size":6},"modified":{"name":"sym.imp.strncmp","addr":4198608,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp._exit","addr":4198624,"size":6},"modified":{"name":"sym.imp._exit","addr":4198624,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.strcpy","addr":4198640,"size":6},"modified":{"name":"sym.imp.strcpy","addr":4198640,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__fpending","addr":4198656,"size":6},"modified":{"name":"sym.imp.__fpending","addr":4198656,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.textdomain","addr":4198672,"size":6},"modified":{"name":"sym.imp.textdomain","addr":4198672,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.fclose","addr":4198688,"size":6},"modified":{"name":"sym.imp.fclose","addr":4198688,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.bindtextdomain","addr":4198704,"size":6},"modified":{"name":"sym.imp.bindtextdomain","addr":4198704,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.dcgettext","addr":4198720,"size":6},"modified":{"name":"sym.imp.dcgettext","addr":4198720,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__ctype_get_mb_cur_max","addr":4198736,"size":6},"modified":{"name":"sym.imp.__ctype_get_mb_cur_max","addr":4198736,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.strlen","addr":4198752,"size":6},"modified":{"name":"sym.imp.strlen","addr":4198752,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__stack_chk_fail","addr":4198768,"size":6},"modified":{"name":"sym.imp.__stack_chk_fail","addr":4198768,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.strrchr","addr":4198800,"size":6},"modified":{"name":"sym.imp.strrchr","addr":4198800,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.lseek","addr":4198816,"size":6},"modified":{"name":"sym.imp.lseek","addr":4198816,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.memset","addr":4198832,"size":6},"modified":{"name":"sym.imp.memset","addr":4198832,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.fscanf","addr":4198848,"size":6},"modified":{"name":"sym.imp.fscanf","addr":4198848,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.close","addr":4198864,"size":6},"modified":{"name":"sym.imp.close","addr":4198864,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__libc_start_main","addr":4198880,"size":6},"modified":{"name":"sym.imp.__libc_start_main","addr":4198880,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.memcmp","addr":4198896,"size":6},"modified":{"name":"sym.imp.memcmp","addr":4198896,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.fputs_unlocked","addr":4198912,"size":6},"modified":{"name":"sym.imp.fputs_unlocked","addr":4198912,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.calloc","addr":4198928,"size":6},"modified":{"name":"sym.imp.calloc","addr":4198928,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.strcmp","addr":4198944,"size":6},"modified":{"name":"sym.imp.strcmp","addr":4198944,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"loc.imp.__gmon_start","addr":4198960,"size":6},"modified":{"name":"loc.imp.__gmon_start","addr":4198960,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.memcpy","addr":4198976,"size":6},"modified":{"name":"sym.imp.memcpy","addr":4198976,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.fileno","addr":4198992,"size":6},"modified":{"name":"sym.imp.fileno","addr":4198992,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.malloc","addr":4199008,"size":6},"modified":{"name":"sym.imp.malloc","addr":4199008,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.nl_langinfo","addr":4199040,"size":6},"modified":{"name":"sym.imp.nl_langinfo","addr":4199040,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.ungetc","addr":4199056,"size":6},"modified":{"name":"sym.imp.ungetc","addr":4199056,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__freading","addr":4199072,"size":6},"modified":{"name":"sym.imp.__freading","addr":4199072,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.realloc","addr":4199088,"size":6},"modified":{"name":"sym.imp.realloc","addr":4199088,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.fdopen","addr":4199104,"size":6},"modified":{"name":"sym.imp.fdopen","addr":4199104,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.setlocale","addr":4199120,"size":6},"modified":{"name":"sym.imp.setlocale","addr":4199120,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__printf_chk","addr":4199136,"size":6},"modified":{"name":"sym.imp.__printf_chk","addr":4199136,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.error","addr":4199152,"size":6},"modified":{"name":"sym.imp.error","addr":4199152,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.open","addr":4199168,"size":6},"modified":{"name":"sym.imp.open","addr":4199168,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.exit","addr":4199216,"size":6},"modified":{"name":"sym.imp.exit","addr":4199216,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.fwrite","addr":4199232,"size":6},"modified":{"name":"sym.imp.fwrite","addr":4199232,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__fprintf_chk","addr":4199248,"size":6},"modified":{"name":"sym.imp.__fprintf_chk","addr":4199248,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.mbsinit","addr":4199264,"size":6},"modified":{"name":"sym.imp.mbsinit","addr":4199264,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.iswprint","addr":4199280,"size":6},"modified":{"name":"sym.imp.iswprint","addr":4199280,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"sym.imp.__ctype_b_loc","addr":4199296,"size":6},"modified":{"name":"sym.imp.__ctype_b_loc","addr":4199296,"size":6}},{"distance":1.000000,"type":"MATCH","original":{"name":"fcn.00401460","addr":4199520,"size":41},"modified":{"name":"fcn.00401470","addr":4199536,"size":41}},{"distance":0.879195,"type":"SIMILAR","original":{"name":"fcn.00401530","addr":4199728,"size":745},"modified":{"name":"fcn.00401540","addr":4199744,"size":745}},{"distance":0.931973,"type":"SIMILAR","original":{"name":"fcn.004018f0","addr":4200688,"size":147},"modified":{"name":"fcn.00401900","addr":4200704,"size":147}},{"distance":0.985401,"type":"SIMILAR","original":{"name":"fcn.00401990","addr":4200848,"size":137},"modified":{"name":"fcn.004019a0","addr":4200864,"size":137}},{"distance":1.000000,"type":"MATCH","original":{"name":"fcn.00401a20","addr":4200992,"size":162},"modified":{"name":"fcn.00401a30","addr":4201008,"size":162}},{"distance":0.964912,"type":"SIMILAR","original":{"name":"fcn.00401ae0","addr":4201184,"size":228},"modified":{"name":"fcn.00401af0","addr":4201200,"size":228}},{"distance":0.991569,"type":"SIMILAR","original":{"name":"fcn.00401be0","addr":4201440,"size":2728},"modified":{"name":"fcn.00401bf0","addr":4201456,"size":2728}},{"distance":0.971765,"type":"SIMILAR","original":{"name":"fcn.00402730","addr":4204336,"size":425},"modified":{"name":"fcn.00402740","addr":4204352,"size":425}},{"distance":0.980000,"type":"SIMILAR","original":{"name":"fcn.004029a0","addr":4204960,"size":50},"modified":{"name":"fcn.004029b0","addr":4204976,"size":50}},{"distance":0.960784,"type":"SIMILAR","original":{"name":"fcn.00402e20","addr":4206112,"size":204},"modified":{"name":"fcn.00402e30","addr":4206128,"size":204}},{"distance":0.952331,"type":"SIMILAR","original":{"name":"fcn.00402fc0","addr":4206528,"size":944},"modified":{"name":"fcn.00402fd0","addr":4206544,"size":944}},{"distance":0.992366,"type":"SIMILAR","original":{"name":"fcn.004033d0","addr":4207568,"size":131},"modified":{"name":"fcn.004033e0","addr":4207584,"size":131}},{"distance":0.994012,"type":"SIMILAR","original":{"name":"fcn.00403460","addr":4207712,"size":167},"modified":{"name":"fcn.00403470","addr":4207728,"size":167}},{"distance":0.961538,"type":"SIMILAR","original":{"name":"fcn.00403590","addr":4208016,"size":26},"modified":{"name":"fcn.004035a0","addr":4208032,"size":26}},{"distance":0.958333,"type":"SIMILAR","original":{"name":"fcn.004035e0","addr":4208096,"size":48},"modified":{"name":"fcn.004035f0","addr":4208112,"size":48}},{"distance":0.975000,"type":"SIMILAR","original":{"name":"fcn.00403750","addr":4208464,"size":40},"modified":{"name":"fcn.00403760","addr":4208480,"size":40}},{"distance":0.857143,"type":"SIMILAR","original":{"name":"fcn.004037a0","addr":4208544,"size":49},"modified":{"name":"fcn.004037b0","addr":4208560,"size":49}},{"distance":0.914286,"type":"SIMILAR","original":{"name":"fcn.004037e0","addr":4208608,"size":35},"modified":{"name":"fcn.004037f0","addr":4208624,"size":35}},{"distance":1.000000,"type":"MATCH","original":{"name":"fcn.00403810","addr":4208656,"size":67},"modified":{"name":"fcn.00403820","addr":4208672,"size":67}},{"distance":0.972222,"type":"SIMILAR","original":{"name":"fcn.00403860","addr":4208736,"size":108},"modified":{"name":"fcn.00403870","addr":4208752,"size":108}},{"distance":0.958621,"type":"SIMILAR","original":{"name":"fcn.004038e0","addr":4208864,"size":1015},"modified":{"name":"fcn.004038f0","addr":4208880,"size":1015}},{"distance":0.932203,"type":"SIMILAR","original":{"name":"fcn.00403cf0","addr":4209904,"size":118},"modified":{"name":"fcn.00403d00","addr":4209920,"size":118}},{"distance":0.949153,"type":"SIMILAR","original":{"name":"fcn.00403d70","addr":4210032,"size":59},"modified":{"name":"fcn.00403d80","addr":4210048,"size":59}},{"distance":0.965116,"type":"SIMILAR","original":{"name":"fcn.00403db0","addr":4210096,"size":86},"modified":{"name":"fcn.00403dc0","addr":4210112,"size":86}},{"distance":0.923077,"type":"SIMILAR","original":{"name":"fcn.00403e90","addr":4210320,"size":26},"modified":{"name":"fcn.00403ea0","addr":4210336,"size":26}}]

[XX] db/tools/rz_diff rz-diff functions comparison
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc '!rz-diff -C -t functions bins/other/rz-diff/true bins/other/rz-diff/false' =
-- stdout
--- expected
+++ actual
@@ -9,6 +9,7 @@
                 sym.imp.strcpy    6 0x00000000004010f0 | MATCH   (1.000000) | 0x00000000004010f0     6 sym.imp.strcpy
             sym.imp.__fpending    6 0x0000000000401100 | MATCH   (1.000000) | 0x0000000000401100     6 sym.imp.__fpending
             sym.imp.textdomain    6 0x0000000000401110 | MATCH   (1.000000) | 0x0000000000401110     6 sym.imp.textdomain
+                sym.imp.fclose    6 0x0000000000401120 | MATCH   (1.000000) | 0x0000000000401120     6 sym.imp.fclose
         sym.imp.bindtextdomain    6 0x0000000000401130 | MATCH   (1.000000) | 0x0000000000401130     6 sym.imp.bindtextdomain
              sym.imp.dcgettext    6 0x0000000000401140 | MATCH   (1.000000) | 0x0000000000401140     6 sym.imp.dcgettext
 sym.imp.__ctype_get_mb_cur_max    6 0x0000000000401150 | MATCH   (1.000000) | 0x0000000000401150     6 sym.imp.__ctype_get_mb_cur_max

@@ -60,11 +61,11 @@
                   fcn.004035e0   48 0x00000000004035e0 | SIMILAR (0.958333) | 0x00000000004035f0    48 fcn.004035f0
                   fcn.00403750   40 0x0000000000403750 | SIMILAR (0.975000) | 0x0000000000403760    40 fcn.00403760
                   fcn.004037a0   49 0x00000000004037a0 | SIMILAR (0.857143) | 0x00000000004037b0    49 fcn.004037b0
-                  fcn.004037e0   41 0x00000000004037e0 | SIMILAR (0.926829) | 0x00000000004037f0    41 fcn.004037f0
+                  fcn.004037e0   35 0x00000000004037e0 | SIMILAR (0.914286) | 0x00000000004037f0    35 fcn.004037f0
                   fcn.00403810   67 0x0000000000403810 | MATCH   (1.000000) | 0x0000000000403820    67 fcn.00403820
                   fcn.00403860  108 0x0000000000403860 | SIMILAR (0.972222) | 0x0000000000403870   108 fcn.00403870
                   fcn.004038e0 1015 0x00000000004038e0 | SIMILAR (0.958621) | 0x00000000004038f0  1015 fcn.004038f0
-                  fcn.00403cf0  124 0x0000000000403cf0 | SIMILAR (0.935484) | 0x0000000000403d00   124 fcn.00403d00
-                  fcn.00403d70   65 0x0000000000403d70 | SIMILAR (0.953846) | 0x0000000000403d80    65 fcn.00403d80
-                  fcn.00403db0   92 0x0000000000403db0 | SIMILAR (0.967391) | 0x0000000000403dc0    92 fcn.00403dc0
-                  fcn.00403e90   32 0x0000000000403e90 | SIMILAR (0.937500) | 0x0000000000403ea0    32 fcn.00403ea0
+                  fcn.00403cf0  118 0x0000000000403cf0 | SIMILAR (0.932203) | 0x0000000000403d00   118 fcn.00403d00
+                  fcn.00403d70   59 0x0000000000403d70 | SIMILAR (0.949153) | 0x0000000000403d80    59 fcn.00403d80
+                  fcn.00403db0   86 0x0000000000403db0 | SIMILAR (0.965116) | 0x0000000000403dc0    86 fcn.00403dc0
+                  fcn.00403e90   26 0x0000000000403e90 | SIMILAR (0.923077) | 0x0000000000403ea0    26 fcn.00403ea0
```

**Review by wargio — APPROVED** [review]

**Comment by XVilka** [comment]
@wargio you haven't said if those tests results are better now, or it's a regression?

**Comment by wargio** [comment]
looks better for me @XVilka

File: librz/debug/p/native/windows/windows_debug.c [hunk 1]
```diff
@@ -424,7 +424,15 @@ int w32_reg_read(RzDebug *dbg, int type, ut8 *buf, int size) {
...
-	HANDLE th = wrap->pi.hThread;
+	HANDLE th = NULL;
+	if (wrap->pi.dwThreadId == dbg->tid) {
+		th = wrap->pi.hThread;
+	} else {
+		PTHREAD_ITEM thread = find_thread(dbg, dbg->tid);
+		if (thread) {
+			th = thread->hThread;
+		}
+	}
```
Comment (ret2libc) [inline librz/debug/p/native/windows/windows_debug.c]: opportunity to extract this into a `get_handle_from_hThread` function?

File: librz/core/cfile.c [hunk 2]
```diff
@@ -697,6 +697,12 @@ static int rz_core_file_do_load_for_debug(RzCore *r, ut64 baseaddr, const char *
 #endif
 	}
 	rz_core_bin_apply_all_info(r, binfile);
+	if (r->dbg->cur->reg_profile) {
+		char *p = r->dbg->cur->reg_profile(r->dbg);
+		rz_reg_set_profile_string(r->dbg->reg, p);
+		rz_debug_reg_sync(r->dbg, RZ_REG_TYPE_ALL, 0);
+		free(p);
```
Comment (ret2libc) [inline librz/core/cfile.c]: shall this be done similar to what's done in https://github.com/rizinorg/rizin/pull/1359/commits/1bf66da0b25a9a586303f017df31002f45ec3738 ? Also, in that piece of code there is no `rz_debug_reg_sync`, is that ok? Maybe we should have a single function to perform this "set_profile_sync" operation, if there is not one already.

**Review by XVilka — APPROVED** [review]
Ok, waiting for @ret2libc to merge.

**Review by ret2libc — APPROVED** [review]

**Pull request merged**